src/tfr_control/src/robot_interface.cpp

```diff
@@ -118,9 +124,10 @@ namespace tfr_control
     void RobotInterface::write() 
     {
         //Grab the neccessary data
-        tfr_msgs::ArduinoReading reading;
-        if (latest_arduino != nullptr)
-            reading = *latest_arduino;
+        tfr_msgs::ArduinoAReading reading_a;
+        tfr_msgs::ArduinoBReading reading_b;
+        if (latest_arduino_a != nullptr)
+            reading_a = *latest_arduino_a;
 
         double signal;
         if (use_fake_values) //test code  for working with rviz simulator
```

Comment:
I'm looking forward to the day where we can finally delete this `if` clause here.

Reply:
Lol I think we all are.